core/sequencer/server.go

```diff
@@ -445,20 +436,29 @@ func (s *Server) HighWatermarks(ctx context.Context, directoryID string, starts
 	}
 	// TODO(gbelvin): Get HighWatermarks in parallel.
 	for _, logID := range logIDs {
-		start := starts[logID]
-		if batchSize <= 0 {
-			watermarks[logID] = start
-			continue
-		}
-		count, high, err := s.logs.HighWatermark(ctx, directoryID, logID, start, batchSize)
+		count, high, err := s.logs.HighWatermark(ctx, directoryID, logID, ends[logID], batchSize)
 		if err != nil {
 			return 0, nil, status.Errorf(codes.Internal,
 				"HighWatermark(%v/%v, start: %v, batch: %v): %v",
-				directoryID, logID, start, batchSize, err)
+				directoryID, logID, ends[logID], batchSize, err)
 		}
-		watermarks[logID] = high
+		starts[logID] = ends[logID]
```

Comment:
Then you could write this like:
`starts[logID], ends[logID] = low, high`

Reply:
oooh, fancy :-)